### ISSUE TITLE
[Epic] Add a LeRobot learning-data plane and guarded StarVLA policy integration

### ISSUE THREAD

**Issue by Quchaosheng** [Quchaosheng/workbench-mobile-home-robot#345]
## Context

Workbench already has a strong evidence-first control plane: bounded `SemanticAction` tools, fail-closed policy validation, append-only events, deterministic replay, and a verifier that distinguishes `confirmed`, `refuted`, and `insufficient_evidence`.

What is still missing is a learning plane for high-rate robot episodes and a controlled way to evaluate learned policies. Two upstream projects provide useful reference designs:

- [Hugging Face LeRobot](https://github.com/huggingface/lerobot): robot/camera/teleoperator interfaces, synchronized episode recording, LeRobotDataset v3, policy training, evaluation, and replay tooling.
- [StarVLA](https://github.com/starVLA/starVLA): modular VLA frameworks, interchangeable backbones/action heads, LeRobot-compatible data loading, checkpoint normalization metadata, and client/server policy inference.

Workbench must adopt these capabilities without weakening its existing authority boundaries. A learned policy is a proposal source or a bounded skill implementation; it is never the verifier, safety authority, emergency-stop authority, or source of physical completion truth.

This work is deliberately P2, consistent with `docs/AI-NATIVE-PLAYBOOK.md`, which defers VLA joint control until simpler runtime and evidence foundations demonstrate a concrete need.

## Objective

Add a LeRobot-compatible learning-data plane and a guarded, pluggable policy interface that can evaluate LeRobot and StarVLA policies in shadow mode before any owner-approved physical execution.

```text
camera / robot state / teleoperation
            |
            v
LeRobot-compatible episode dataset ---> offline training / StarVLA policy service
                                                    |
                                                    v
                                             PolicyProposal
                                                    |
user goal -> TaskGraph -> SemanticAction -> trusted skill boundary
                                                    |
                                     validation / limits / Motion / MCU
                                                    |
                                  WorldEvent -> verifier -> replay
```

## Architectural rules

1. High-rate images, states, and actions belong to a separate learning-data plane; they must not be embedded in the SQLite `WorldEvent` store.
2. World events retain bounded metadata, hashes, time ranges, provenance, and evidence references that point to learning artifacts.
3. Record proposed, safety-adjusted, and actually executed actions separately. Training targets must never silently use an unexecuted proposal.
4. Only a Workbench verifier may declare task success. `ActionResult.outcome=completed` and a successful model response are not sufficient.
5. A policy response must not become raw joint, velocity, firmware, safe-enable, emergency-stop, or completion authority through the Agent Runtime.
6. Physical policy execution is out of scope until a separate owner-approved Motion/Safety task exists.
7. Exact camera order, action/state layout, units, frame IDs, calibration revision, normalization statistics, model digest, dataset digest, code revision, control rate, and action horizon must be versioned and checked fail-closed.

## Milestone 0 — design and dependency review

- [ ] Add an ADR defining the learning plane, policy proposal boundary, evidence links, and shutdown behavior.
- [ ] Add LeRobot and StarVLA to `THIRD_PARTY_REVIEW.md` with code, model-weight, dataset, and checkpoint licenses reviewed separately.
- [ ] Review StarVLA's repository license text before vendoring any source; prefer an adapter/process boundary over copied code.
- [ ] Define an exit path that leaves the deterministic scripted runner and existing verifier fully usable when ML dependencies are absent.

## Milestone 1 — LeRobot-compatible dataset export

Create an optional integration, initially outside shared `interfaces/`, that exports or records episodes containing at least:

```text
observation.images.<camera>
observation.state.<group>
action.proposed
action.commanded
action.executed
task
timestamp
```

Episode metadata should include:

- `run_id`, `scenario_id`, source/evidence class and software commit;
- robot, kinematic, calibration and dataset schema revisions;
- camera names/order, state/action keys, units, frame IDs and FPS;
- artifact hashes and references to applicable `ActionResult` and `VerificationResult` records;
- terminal verification status and reason code.

Acceptance criteria:

- [ ] Export is deterministic for the same bounded input artifact.
- [ ] Images/video remain outside SQLite; event records contain only bounded references and hashes.
- [ ] Schema/layout mismatch, missing calibration, non-finite values, duplicate samples, timestamp regression, or incomplete action provenance fails closed.
- [ ] `confirmed`, `refuted`, and `insufficient_evidence` episodes remain distinguishable.
- [ ] Default imitation-learning selection includes only explicitly eligible executed/verified samples; failed episodes remain available for analysis without becoming positive targets.
- [ ] Unit and fixture tests run without GPU, Hub credentials, or network access.

## Milestone 2 — policy provider and manifest

Introduce a narrow optional policy port, for example:

```python
class PolicyProvider(Protocol):
    def predict_action(self, observation, instruction) -> PolicyProposal: ...
```

Initial providers:

- [ ] deterministic fixture provider;
- [ ] local/in-process LeRobot policy adapter;
- [ ] remote StarVLA-compatible policy adapter.

Every loaded policy must have a validated manifest containing model/checkpoint digest, training dataset digest, code revision, camera contract, state/action key order and dimensions, coordinate conventions, normalization revision/statistics, action horizon, control rate, supported skill, and resource requirements.

Acceptance criteria:

- [ ] Unknown or mismatched metadata fails closed before inference.
- [ ] Requests have bounded payload size, deadlines, request IDs and cancellation behavior.
- [ ] Remote endpoints default to localhost/private deployment boundaries and never return raw tracebacks to untrusted clients.
- [ ] Timeout, malformed output, NaN/Inf, wrong shape, stale observation and unsupported action space produce typed rejection evidence.
- [ ] ML packages remain optional and cannot break the offline canonical demo.

## Milestone 3 — shadow-mode evaluation

Add a policy evaluation runner that reuses frozen scenarios and the existing verifier while preventing policy output from reaching Motion or MCU.

Required metrics:

- verifier-confirmed, refuted and insufficient-evidence rates;
- policy rejection and malformed-output rates;
- safe-stop/timeout counters where applicable;
- inference latency P50/P95 and missed control deadlines;
- difference between proposed, commanded and executed actions;
- per-scenario, per-seed and per-policy-manifest results.

Acceptance criteria:

- [ ] Shadow mode is mechanically unable to dispatch physical control.
- [ ] Same scenario/seed inputs can compare scripted, LeRobot and StarVLA providers.
- [ ] Reports carry `run_id`, policy manifest digest, evidence references and truthful `executed`/`release_eligible` fields.
- [ ] A model may not count its own confidence or response status as task success.

## Milestone 4 — separately approved bounded skill pilot

After the preceding gates pass, open a separate owner-approved Issue for one bounded skill such as known-object grasping in simulation/HIL.

Promotion evidence should include:

- [ ] an explicit semantic action and fixed action-space boundary;
- [ ] Motion/Safety owner approval;
- [ ] limits, watchdog, cancellation and safe-stop tests;
- [ ] held-out scenario results and regression comparison against the deterministic baseline;
- [ ] post-action observation and Workbench verification;
- [ ] no raw policy path to MCU, E-stop, safe-enable or completion claims.

## Non-goals

- Online learning or self-modifying policies.
- Direct VLA joint control from `services/agent_runtime/`.
- Replacing Workbench events, replay, policy validation or verifier with LeRobot/StarVLA abstractions.
- Direct physical replay of dataset actions.
- Making PyTorch, CUDA, Transformers, LeRobot or StarVLA mandatory dependencies for the core offline runtime.
- Claiming real-robot capability from scripted or shadow-mode results.

## Suggested initial paths

```text
integrations/lerobot/
services/policy_runtime/
evaluation/policy/
tests/integration/
docs/decisions/
```

Shared `interfaces/`, `robot/control/`, firmware and physical deployment paths are not part of the first implementation slice. Any later change to those paths requires the repository's existing approvals and validation rules.

## Related Workbench issues

- #151 — require a post-action observe and verify loop
- #158 — produce real RGB-D observations
- #302 — bind run artifacts, replay and verification to scenario identity
- #304 — validate scenarios against safety and evidence gates
- #309 — multi-scenario evidence-first runtime epic
- #313 — deterministic time, randomness and environment provenance
- #317 — trace user problems to scenario and run evidence

## Definition of done for this Epic

This Epic is complete only when Workbench can deterministically produce a validated learning episode, load at least one optional learned policy through a manifest-checked provider, run it in non-dispatching shadow evaluation against frozen scenarios, and generate evidence-linked reports whose success status comes exclusively from the existing Workbench verifier. Physical learned-policy execution is explicitly not required to close this Epic.

**Comment by Quchaosheng** [comment]
Epic closed on owner request. Milestone 0 landed as PR #378 (commit 1dbc1b7): docs/decisions/ADR-0008-learning-data-plane.md and the guarded policy interface. Later milestones of this Epic are not claimed as complete.
